Objects/stringlib/unicode_format.h

```diff
@@ -1042,16 +1061,26 @@ formatteriter_next(formatteriterobject *it)
             conversion_str = PyUnicode_FromKindAndData(PyUnicode_4BYTE_KIND,
                                                        &conversion, 1);
         if (conversion_str == NULL)
-            goto done;
+            goto error;
 
-        tuple = PyTuple_Pack(4, literal_str, field_name_str, format_spec_str,
-                             conversion_str);
-    done:
+        Py_INCREF((PyObject *) &FormatterIterResultType);
+        res = PyStructSequence_New(&FormatterIterResultType);
+
+        if (res == NULL)
+            goto error;
+
+        PyStructSequence_SET_ITEM(res, 0, literal_str); 
+        PyStructSequence_SET_ITEM(res, 1, field_name_str);
+        PyStructSequence_SET_ITEM(res, 2, format_spec_str); 
+        PyStructSequence_SET_ITEM(res, 3, conversion_str); 
+
+        return res;
+    error:
         Py_XDECREF(literal_str);
         Py_XDECREF(field_name_str);
         Py_XDECREF(format_spec_str);
         Py_XDECREF(conversion_str);
-        return tuple;
+        return res;
```

Comment:
Just `return NULL`. And initializing `res` with NULL at line 1035 is not needed.